### PR TITLE
hw21:patterns: abstract factory, decorator, adapter, proxy

### PR DESCRIPTION
**Pull request by u2107680007911543774e7r**

### CODE REVIEW

**Review by MateAcademy — APPROVED** [review]
Good homework!!!!!!!!!!!!!!!!!!!!!!

File: src/main/java/NotebookFactory/ChineseMcBook.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package NotebookFactory;
+
+public class ChineseMcBook extends Notebook {
```
Comment (SerhiiPasko) [inline src/main/java/NotebookFactory/ChineseMcBook.java:3]: Notebook = блокнот. Laptop it is correct translation =)

File: src/main/java/NotebookFactory/NotebookStore.java [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package NotebookFactory;
+
+public abstract class NotebookStore {
+
+	public Notebook prepareForSaleNotebook(String type) {
```
Comment (SerhiiPasko) [inline src/main/java/NotebookFactory/NotebookStore.java:5]: Template detected

File: src/main/java/NotebookFactory/ShanghaiNotebookStore.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package NotebookFactory;
+
+public class ShanghaiNotebookStore extends NotebookStore{
```
Comment (SerhiiPasko) [inline src/main/java/NotebookFactory/ShanghaiNotebookStore.java:3]: Where you use it (main or test)? Also I don't see abstract factory. Read https://www.baeldung.com/java-abstract-factory-pattern

File: src/main/java/PhoneDecorator/SamsungPhone.java [hunk 4]
```diff
@@ -0,0 +1,14 @@
+package PhoneDecorator;
+
+public class SamsungPhone extends Phone{
```
Comment (SerhiiPasko) [inline src/main/java/PhoneDecorator/SamsungPhone.java:3]: Also. How you cheack, that you wrote correctly?

File: src/main/java/ProjectProxy/ProjectProxy.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+	private RealProject real;
+	private String name;
+
+	public ProjectProxy(String name) {
+		this.name = name;
+	}
+
+	@Override
+	public void run() {
+		if (real.equals(null)) {
```
Comment (SerhiiPasko) [inline src/main/java/ProjectProxy/ProjectProxy.java:13]: And what that proxy give for us?

**Review by SerhiiPasko — CHANGES_REQUESTED** [review]
Read comments. I don't see usages of this patterns (main methods or tests)

File: src/main/java/NotebookFactory/ChineseMcBook.java (hunk 1 above)
Comment (u2107680007911543774e7r) [reply src/main/java/NotebookFactory/ChineseMcBook.java:3]: who said that i need laptop, its notebook factory by apple...

File: src/main/java/NotebookFactory/ShanghaiNotebookStore.java (hunk 3 above)
Comment (u2107680007911543774e7r) [reply src/main/java/NotebookFactory/ShanghaiNotebookStore.java:3]: I deleted all mains since it's not part of pattern

**Pull request merged**